torchgeo/datasets/ftw.py

```diff
@@ -0,0 +1,363 @@
+                    ),
+                }
+                files.append(sample)
+
+        return files
+
+    def _load_image(self, path: Path) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the loaded image
+        """
+        filename = os.path.join(path)
+        with rasterio.open(filename) as f:
+            array: np.typing.NDArray[np.int_] = f.read()
+            tensor = array_to_tensor(array).float()
+        return tensor
+
+    def _load_target(self, path: Path) -> Tensor:
+        """Load a single mask corresponding to image.
+
+        Args:
+            path: path to the mask
+
+        Returns:
+            the mask of the image
+        """
+        filename = os.path.join(path)
+        with rasterio.open(filename) as f:
+            array: np.typing.NDArray[np.int_] = f.read(1)
+            tensor = torch.from_numpy(array).long()
+        return tensor
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset."""
+        for country in self.countries:
+            if self._verify_data(country):
+                continue
+
+            filename = f'{country}.zip'
+            pathname = os.path.join(self.root, filename)
+            if os.path.exists(pathname):
+                extract_archive(pathname, os.path.join(self.root, country))
+                continue
+
+            if not self.download:
+                raise DatasetNotFoundError(self)
+
+            download_and_extract_archive(
+                self.base_url + filename,
+                os.path.join(self.root, country),
+                filename=filename,
+                md5=self.country_to_md5[country] if self.checksum else None,
+            )
+
+    def _verify_data(self, country: str) -> bool:
+        """Verify that data for a country is extracted.
+
+        Args:
+            country: the country to check
+
+        Returns:
+            True if the dataset directories and split files are found, else False
+        """
+        for entry in ['label_masks', 's2_images', f'chips_{country}.parquet']:
+            if not os.path.exists(os.path.join(self.root, country, entry)):
+                return False
+
+        return True
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: str | None = None,
+    ) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample return by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional suptitle to use for figure
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        fig, axs = plt.subplots(nrows=1, ncols=3, figsize=(15, 5))
+
+        win_a = sample['image'][0:3].permute(1, 2, 0)
+        win_b = sample['image'][4:7].permute(1, 2, 0)
```

Comment:
I'm starting to want to use einops in places like this to make it more clear exactly what's happening. Optional request

Reply:
Done